saleor/graphql/order/tests/test_order.py

```diff
@@ -9159,6 +9159,58 @@ def test_orders_query_with_filter_charge_status(
     assert content["data"]["orders"]["totalCount"] == expected_count
 
 
+def test_order_query_with_filter_numbers(
+    orders_query_with_filter,
+    staff_api_client,
+    permission_manage_orders,
+    orders,
+    channel_USD,
+):
+    # given
+    variables = {
+        "filter": {
+            "numbers": [orders[0].number, orders[2].number],
+        },
+    }
+
+    # when
+    response = staff_api_client.post_graphql(
+        orders_query_with_filter, variables, permissions=(permission_manage_orders,)
+    )
+
+    # then
+    content = get_graphql_content(response)
+    orders = content["data"]["orders"]["edges"]
+    assert len(orders) == 2
```

Comment:
I would also check that the order numbers we got in the response are in fact the same order numbers we passed to the filter.